tests/bud.bats

```diff
@@ -202,6 +202,202 @@ _EOF
+RUN echo world > hello
+
+# hello should get copied since we are accessing stage from its numeric value and not
+# addtional build context where some-stage is docker://alpine
+FROM alpine
+COPY --from=0 hello .
+RUN cat hello
+_EOF
+
+  run_buildah build $WITH_POLICY_JSON --build-context some-stage=docker://alpine -t test -f ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile2
+  expect_output --substring "world"
+}
+
+# Test conflict between stage short name and additional-context conflict on FROM
+# Buildkit parity give priority to additional-context over stage names.
+@test "build-with-additional-build-context and FROM, stagename and additional-context conflict" {
+  mkdir -p ${TEST_SCRATCH_DIR}/bud/platform
+
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile1 << _EOF
+FROM alpine
+RUN touch hello
+RUN echo world > hello
+_EOF
+
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile2 << _EOF
+FROM alpine as some-stage
+RUN echo world
+
+# hello should be there since we are giving priority to additional context
+FROM some-stage
+RUN cat hello
+_EOF
+
+  # Build a first image which we can use as source
+  run_buildah build $WITH_POLICY_JSON -t source -f ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile1
+  # Second FROM should choose base as `source` instead of local-stage named `some-stage`.
+  run_buildah build $WITH_POLICY_JSON --build-context some-stage=docker://source -t test -f ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile2
+  expect_output --substring "world"
+}
+
+# Test adding additional build context
+@test "build-with-additional-build-context and COPY, additional context from host" {
+  mkdir -p ${TEST_SCRATCH_DIR}/bud/platform
+  mkdir -p ${TEST_SCRATCH_DIR}/bud/platform2
+
+  # add file on original context
+  echo something > ${TEST_SCRATCH_DIR}/bud/platform/somefile
+  # add file on additional context
+  echo hello > ${TEST_SCRATCH_DIR}/bud/platform2/hello
+
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile << _EOF
+FROM alpine
+COPY somefile .
+RUN cat somefile
+COPY --from=context2 hello .
+RUN cat hello
+_EOF
+
+  # Test additional context
+  run_buildah build $WITH_POLICY_JSON -t source --build-context context2=${TEST_SCRATCH_DIR}/bud/platform2 ${TEST_SCRATCH_DIR}/bud/platform
+  expect_output --substring "something"
+  expect_output --substring "hello"
+}
+
+# Test adding additional build context but download tar
+@test "build-with-additional-build-context and COPY, additional context from external URL" {
+  mkdir -p ${TEST_SCRATCH_DIR}/bud/platform
+
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile << _EOF
+FROM alpine
+COPY --from=crun-context . .
+RUN ls -a
+_EOF
+
+  # Test additional context but download from tar
+  run_buildah build $WITH_POLICY_JSON -t source --build-context crun-context=https://github.com/containers/crun/releases/download/1.4.5/crun-1.4.5.tar.xz ${TEST_SCRATCH_DIR}/bud/platform
+  # additional context from tar must show crun binary inside container
+  expect_output --substring "crun-1.4.5"
```

Comment:
This should be checking for a pattern that only matches the name of a file in the archive, and not a pattern that is also part of the archive's name, in case we break something and stop extracting its contents in the future.

Reply:
now checking for specific files `libcrun`.

Reply:
now checking for specific files `libcrun`.